src/resdk/tables/variant.py

```diff
@@ -0,0 +1,157 @@
+""".. Ignore pydocstyle D400.
+
+=============
+VariantTables
+=============
+
+.. autoclass:: VariantTables
+    :members:
+
+"""
+import re
+from functools import lru_cache
+
+import pandas as pd
+
+from .base import BaseTables
+
+
+class VariantTables(BaseTables):
+    """A helper class to fetch collection's variant, qc and meta data.
+
+    This class enables fetching given collection's data and returning it
+    as tables which have samples in rows and variants / qc / metadata
+    in columns.
+
+    A simple example:
+
+    .. code-block:: python
+
+        # Get Collection object
+        collection = res.collection.get("collection-slug")
+
+        # Fetch collection variant data
+        tables = VariantTables(collection)
+        var = tables.variants
+
+    """
+
+    process_type = "data:mutationstable"
+    VARIANTS = "variants"
+    DEPTH = "depth"
+    COVERAGE = "coverage"
+
+    data_type_to_field_name = {
+        VARIANTS: "tsv",
+        DEPTH: "tsv",
+        COVERAGE: "tsv",
+    }
+
+    @property
+    @lru_cache()
+    def mutation_subset_genes(self) -> pd.DataFrame:
+        """Return genes for which variants are reported."""
+        return self._data[0].output["genes"]
```

Comment:
The docstring could be extended because it could be misleading for someone not familiar with `mutations-table`. For example, we use gene `FOO` and its mutation `Val10` as an input to the `mutations-table`. After using the function and getting `["FOO"]` as a list of genes one could naively assume that all other mutations of this gene were considered as well.